### ISSUE TITLE
Error for !ANALYSE !POTENTIAL

### ISSUE THREAD

**Issue by lksrmp** [cp-paw/cp-paw#41]
Running the control file with
```
!ANALYSE
      !DENSITY TITLE='density' FILE='density.wv' TYPE='TOTAL' !END
      !DENSITY TITLE='spin' FILE='spin.wv' TYPE='SPIN' !END
      !POTENTIAL TITLE='potential' FILE='potential.wv' !END
  !END
```
causes and error (see below). Removing `!POTENTIAL ... !END` leads to a normal run. More investigation needed.

Error in protocol file:
```
RUN-TIME REPORT [FRI 19 JUL 2024 10:14 (31.702S) ]
==================================================
NUMBER OF TASKS...............:        36
ELAPSED WALLCLOCK TIME........:     0H 4M28.4S
TOTAL CPU TIME................:     3H 2M20.0S
 
ID                                   TOTAL CPU      CPU/#ITER CPU/TOT
----------------------------------------------------------------------
TOTALCLOCK.....................     3H 2M20.0S     3H 2M20.0S 100 %
INITIALIZATION.................     3H 2M20.0S     3H 2M20.0S 100 %
SETUP CONSTRUCTION.............     0H 0M44.3S     0H 0M44.3S   0 %
SCF-ATOM.......................     0H 0M28.8S     0H 0M28.8S   0 %
MAKEPARTIALWAVES...............     0H 0M 6.7S     0H 0M 6.7S   0 %
TEST SCATTERING................     0H 0M 5.0S     0H 0M 5.0S   0 %
TEST GHOSTS....................     0H 0M 2.7S     0H 0M 2.7S   0 %
BESSELTRANSFORMS...............     0H 0M 0.3S     0H 0M 0.3S   0 %
WAVES__INITIALIZE..............     0H 1M13.7S     0H 1M13.7S   1 %
RESTART I/O....................     3H 0M16.9S     3H 0M16.9S  99 %
MPI ERROR IN MPI_ALLREDUCE
MPI_ERR_TRUNCATE: message truncated
VARIABLE LENG HAS THE VALUE     516000
VARIABLE OPERATION HAS THE VALUE +
VARIABLE CID HAS THE VALUE MONOMER
VARIABLE THISTASK HAS THE VALUE          1
STOP IN MPE__COMBINER8R3
PROCESSOR REQUESTING STOP:   1 OF  36
```

**Comment by lksrmp** [comment]
The size of `ONECPOT` changes during the assignment `ONECPOT=AE1CPOT-PS1CPOT` to the sizes of `AE1CPOT`/`PS1CPOT`. For my test case the size of those in the first dimension is 1032 for tasks 1-4, but 1238 for tasks 5-6. 1238 is then also the size that is allocated.

**Comment by pbloechl** [comment]
One problem is a coding error: In graphics$set1cpot the grid length of an individual atom is provided instead of the maximum radial grid-length for all atoms. As a result the shape and size of ae1cpot depends on the atom, which initiated the allocation. 

The second problem was that an assignment (=) of two allocatable arrays not only changed the values but also the shape of the target array.

Based on this analysis I will try to fix it via branch fix/issue41

**Comment by pbloechl** [comment]
The problem is fixed in pull request #42 named fix/issue41 and is ready for review.
